### DISCUSSION TITLE
Is there a way to reference subdocument?

### DISCUSSION THREAD

**Discussion by vitorhm** [typegoose/typegoose#913] [Q&A] [answered]
Currently I have:

```ts
export class Category {
  
  // ..............

  @typegoose.prop({ _id: false })
  public pizza?: Pizza
}

export class Pizza {

  // ..............

  @typegoose.prop({ type: PizzaSize, required: true, default: [] })
  public sizes!: mongoose.Types.Array<PizzaSize>
}

export class PizzaPrice {

   // ..............

  @typegoose.prop({ required: true, ref: () => PizzaSize })
  public size!: typegoose.Ref<PizzaSize>
}
``` 

`PizzaSize` is a subDocument from `Category.Pizza` and I want `PizzaPrice.size` to reference it.

**Comment by hasezoey** [comment, marked as answer]
to my knowledge it is not possible to reference sub-documents, you will likely need to out-source the things you want to reference to their own collections, or reference the parents model

**Reply by vitorhm** [reply]
I cant reference the parent model because the sizes property is an array, I need to reference one item. I extracted the sizes to a different collection, I think thats a better approach.

Thank you @hasezoey
